runtime/drivers/clickhouse/olap.go

```diff
@@ -206,22 +208,124 @@ func (c *connection) AlterTableColumn(ctx context.Context, tableName, columnName
 }
 
 // CreateTableAsSelect implements drivers.OLAPStore.
-func (c *connection) CreateTableAsSelect(ctx context.Context, name string, view bool, sql string) error {
+func (c *connection) CreateTableAsSelect(ctx context.Context, name string, view bool, sql string, tableOpts map[string]any) error {
 	if view {
 		return c.Exec(ctx, &drivers.Statement{
 			Query:    fmt.Sprintf("CREATE OR REPLACE VIEW %s AS %s", safeSQLName(name), sql),
 			Priority: 100,
 		})
 	}
+	var create strings.Builder
+	create.WriteString("CREATE OR REPLACE TABLE ")
+	create.WriteString(safeSQLName(name))
+	// see if there is any column override
+	if prop, ok := tableOpts["columns"].(string); ok {
+		create.WriteString(prop)
+	}
```

Comment:
See other comment – can we parse into `ModelOutputProperties` here to avoid these type casts?

Reply:
Do you mean we should change `	CreateTableAsSelect(... tableOpts map[string]any) error` to `CreateTableAsSelect(... tableOpts map[string]string) error` because then only we can avoid these type casts ?

And on the caller side generate `tableOpts` from `ModelOutputProperties` ?
```
tableOpts := make(map[string]string)
if err := mapstructure.WeakDecode(outputProps, tableOpts); err != nil {
	return nil, fmt.Errorf("failed to generate tableOpts: %w", err)
}
```

Reply:
No, I just mean that if you add the props in `ModelOutputProperties` as this comment says: https://github.com/rilldata/rill/pull/5083#discussion_r1648170276.

Then in this func you can do something like:
```go
props := &ModelOutputProperties{}
err := mapstructure.WeakDecode(tableOpts, props)
// use props.Columns instead of tableOpts["columns"].(string)
```